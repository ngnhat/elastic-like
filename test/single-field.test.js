const Store = require('../index');

describe('single field', () => {
  it('a single document', () => {
    const store = new Store();
    store.add({ Id: 1, Code: 'aaa', Name: 'quick brown fox' });

    expect(store.search('fox')).toMatchSnapshot();
  });

  it('multiple documents', () => {
    const store = new Store();
    store.add({ Id: 1, Code: '1', Name: 'The quick brown fox' });
    store.add({ Id: 2, Code: '2', Name: 'The quick brown fox jumps over the lazy dog' });
    store.add({ Id: 3, Code: '3', Name: 'The quick brown fox jumps hahaha over the quick dog' });
    store.add({ Id: 4, Code: '4', Name: 'Brown fox hahaha brown dog' });

    expect(store.search('hahaha')).toMatchSnapshot();
  });
});

// { Id: 1, Code: 'aaa', Name: 'aaa bbb ccc' },
// { Id: 2, Code: 'bbb', Name: 'bbb ccc ddd' },
// { Id: 3, Code: 'ccc', Name: 'ccc ddd eee' },
