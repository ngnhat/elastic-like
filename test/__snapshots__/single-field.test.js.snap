// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`single field a single document 1`] = `
Array [
  Object {
    "score": 0.28768207245178085,
    "source": Object {
      "Code": "aaa",
      "Id": 1,
      "Name": "quick brown fox",
    },
  },
]
`;

exports[`single field multiple documents 1`] = `
Array [
  Object {
    "score": 0.7848872485752322,
    "source": Object {
      "Code": "4",
      "Id": 4,
      "Name": "Brown fox hahaha brown dog",
    },
  },
  Object {
    "score": 0.5897495348410585,
    "source": Object {
      "Code": "3",
      "Id": 3,
      "Name": "The quick brown fox jumps hahaha over the quick dog",
    },
  },
]
`;
